### PR TITLE
Favorites from Routes And Stops page

### LINKED ISSUE #209: Add easier way for adding things to favorites
On My Buses, when you click the big ➕ to add a route or stop to favorites, it just redirects to Routes and Stops and then you have to click on one, be taken to its page, and then add it to favorites. This should be easier.

### PR DESCRIPTION
**Pull request by bgregg**
Finally. You can now favorite routes and stops from the routes and stops page directly. Closes #209.

### CODE REVIEW

File: www/directives/route/route-directive.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="row-left">
+    <i class="icon ion-android-bus"></i>
+    <span style="color: #{{route.Color}}; font-size: 125%; font-weight: bold; margin-left: 10px">
+      {{route.RouteAbbreviation}}
+    </span>
+    <p>{{route.GoogleDescription}}</p>
+</div>
+<button ng-class="liked ? 'button button-icon icon ion-ios-heart' : 'button button-icon icon ion-ios-heart-outline'" ng-click="liked = !liked; toggle(route); $event.stopPropagation();"></button>
```
Comment (akaplo) [inline www/directives/route/route-directive.html:9]: Perhaps a comment about `$event.stopPropogation` would be nice here.

File: www/directives/stop/stop-directive.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div class="row-left">
+  <i class="icon ion-ios-location" style="margin-right: 10px"></i>
+  <p>{{stop.Description}} ({{stop.StopId}})</p>
+</div>
+<button ng-class="liked ? 'button button-icon icon ion-ios-heart' : 'button button-icon icon ion-ios-heart-outline'" ng-click="liked = !liked; toggle(stop); $event.stopPropagation();"></button>
```
Comment (akaplo) [inline www/directives/stop/stop-directive.html:6]: Perhaps a comment about `$event.stopPropogation` would be nice here.

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 3]
```diff
@@ -8,43 +8,58 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
   /*
    * Get all the routes and stops
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: This comment doesn't match the function below it.

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 4]
```diff
@@ -8,43 +8,58 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
    */
+
+   $scope.redirectRoute = function(routeId){
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: Comment this and [this](https://github.com/umts/pvta-multiplatform/pull/268/files?diff=unified#diff-365f07089626b424259d271203b2ff1eR13).

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 5]
```diff
@@ -8,43 +8,58 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
      * to us for displaying them.  It makes searching easier.
      */
-    function prepareStops (list) {
-      return _.map(list, function (stop) {
-        return _.pick(stop, 'StopId', 'Description');
-      });
-    }
   }
+
+  function stripDetails (routeList) {
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: We have a function called `prepareStops`, why not rename this to `prepareRoutes?`

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 6]
```diff
@@ -8,43 +8,58 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
-      $scope.stops = prepareStops(stops);
       StopsForage.save(stops);
-      $ionicLoading.hide();
+      getFavoriteStops(stops);
       redraw();
     });
 
+    $ionicLoading.hide();
+
     /* Similar to prepareRoutes, we only
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: Move this comment to `prepareStops`' new location

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 7]
```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
+  function getFavoriteStops (stops){
     localforage.getItem('favoriteStops', function (err, value) {
       $scope.favoriteStops = value;
+      $scope.stops = prepareStops(stops);
       redraw();
     });
-  }
+  };
+
+  $scope.toggleStopHeart = function (stop) {
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js:169]: Comment this

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 8]
```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
+        FavoriteStops.remove(stop);
+      }
+      else {
+        FavoriteStops.push(stop);
+      }
+      $scope.$apply();
+    });
+  };
+
+  $scope.toggleRouteHeart = function(route){
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: Comment this.

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 9]
```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
+    FavoriteRoutes.contains(route, function(bool){
+      if(bool) {
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: Let's change this to be a `===`

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 10]
```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
       $scope.favoriteStops = value;
+      $scope.stops = prepareStops(stops);
       redraw();
     });
-  }
+  };
+
+  $scope.toggleStopHeart = function (stop) {
+    FavoriteStops.contains(stop.StopId, function (bool) {
+      if (bool) {
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: Let's change this to be a `===`

File: www/pages/routes-and-stops/routes-and-stops-controller.js [hunk 11]
```diff
@@ -111,22 +126,53 @@ angular.module('pvta.controllers').controller('RoutesAndStopsController', functi
...
 
   function redraw () {
     $scope.display($scope.currentDisplay);
   }
-  getRoutesAndStops();
+
   $scope.$on('$ionicView.enter', function () {
-    getFavorites();
+    getRoutesAndStops();
+    $scope.display($scope.currentDisplay);
```
Comment (akaplo) [inline www/pages/routes-and-stops/routes-and-stops-controller.js]: We have a [function](https://github.com/umts/pvta-multiplatform/pull/268/files?diff=unified#diff-365f07089626b424259d271203b2ff1eR170) (`redraw`) that does this for us...call it.

**Review by akaplo — CHANGES_REQUESTED** [review]
A few small things, but overall this is fantastic.

**Comment by bgregg** [comment]
The rest of the CodeClimate issues are just "similar code found in other places" and so I think this is good. I also fixed your suggestions @akaplo.

**Comment by bgregg** [comment]
Before:

screen shot 2016-11-18 at 2 07 03 pm

**Comment by bgregg** [comment]
Now:

screen shot 2016-11-18 at 2 06 21 pm

File: www/pages/routes-and-stops/routes-and-stops-controller.js (hunk 3 above)
Comment (akaplo) [reply www/pages/routes-and-stops/routes-and-stops-controller.js]: Looks better now.

**Review by akaplo — APPROVED** [review]

**Comment by akaplo** [comment]
Looks good to me!  Merging.

**Pull request merged**